src/RageSoundMixBuffer.cpp

```diff
@@ -31,19 +31,36 @@ void RageSoundMixBuffer::SetWriteOffset( int iOffset )
 	m_iOffset = iOffset;
 }
 
-void RageSoundMixBuffer::Extend( unsigned iSamples )
+void RageSoundMixBuffer::Extend(unsigned iSamples)
 {
-	const unsigned realsize = iSamples+m_iOffset;
-	if( m_iBufSize < realsize )
+	const std::uint64_t m_nBufSize = static_cast<std::uint64_t>(iSamples) + m_iOffset;
```

Comment:
Style nitpick:
`m_` prefix should be used for member variables. `m_nBufSize` is a local variable, it might be better to omit it.

Also, `m_nBufSize` is really similar to `m_iBufSize`. Maybe something like `iBufSizeNew` would be more readable?